src/sentry_tracing.c

```diff
@@ -431,3 +472,37 @@ sentry_transaction_set_status(
 {
     set_status(tx->inner, status);
 }
+
+static void
+sentry__span_iter_headers(sentry_value_t span,
+    sentry_iter_headers_function_t callback, void *userdata)
+{
+    sentry_value_t trace_id = sentry_value_get_by_key(span, "trace_id");
+    sentry_value_t span_id = sentry_value_get_by_key(span, "span_id");
+    sentry_value_t sampled = sentry_value_get_by_key(span, "sampled");
+
+    if (sentry_value_is_null(trace_id) || sentry_value_is_null(span_id)) {
+        return;
+    }
+
+    char buf[64];
+    snprintf(buf, sizeof(buf), "%s-%s-%s", sentry_value_as_string(trace_id),
+        sentry_value_as_string(span_id),
+        sentry_value_is_true(sampled) ? "1" : "0");
```

Comment:
is this guaranteed not to be more that 64 chars?  I mean really shouldn't the return code of snprintf be checked?

Reply:
pretty much, yes. `trace_id` is 32 chars, `span_id` is 16. Even if for whatever reason they exceed that limit, it wouldn’t be "unsafe", just generate an invalid value. IMO not worth it, we do not check the return value of this in quite a bunch of places.